annotation/check_data.py

```diff
@@ -0,0 +1,132 @@
+
+    # Map image IDs to file names
+    image_id_to_filename = {image_info['id']: "images/" + image_info['file_name'] for image_info in coco_annotations['images']}
+
+    # Initialize predictions dictionary with empty arrays for each class
+    for image_info in coco_annotations['images']:
+        filename = "images/" + image_info['file_name']
+        mmdet_predictions[filename] = [np.empty((0, 5), dtype=np.float32) for _ in range(num_classes)]
+
+    # Populate the predictions with bounding boxes from COCO annotations
+    for annotation in coco_annotations['annotations']:
+        image_id = annotation['image_id']
+        filename = image_id_to_filename[image_id]
+        category_id = annotation['category_id']
+        
+        # Only process category IDs that are within the number of classes
+        if category_id < num_classes:
+            bbox = annotation['bbox']  # COCO bbox is [x_min, y_min, width, height]
+            score = annotation.get('score', 1.0)  # Default score to 1.0 if not provided
+            
+            # Convert COCO bbox to [x_min, y_min, x_max, y_max, score]
+            x_min, y_min, width, height = bbox
+            x_max = x_min + width
+            y_max = y_min + height
+            mmdet_bbox = np.array([x_min, y_min, x_max, y_max, score], dtype=np.float32)
+
+            # Append the bounding box to the corresponding class array
+            mmdet_predictions[filename][category_id] = np.vstack([mmdet_predictions[filename][category_id], mmdet_bbox])
+
+    return mmdet_predictions
+
+
+def compute_mistakes_score(
+    gt_coco_path: str,
+    preds_coco_path: str,
+    classes_mapping: dict[int, str],
+    data_dir: str = None
+):
+    # Load COCO annotations
+    with open(gt_coco_path, 'r') as f:
+        coco_data = json.load(f)
+
+    with open(preds_coco_path, 'r') as f:
+        pred_data = json.load(f)
+
+    labels = coco_to_cleanlab(coco_data)
+    predictions = list(coco_to_mmdet(pred_data, num_classes=2).values())
+    label_issue_idx = find_label_issues(labels, predictions, return_indices_ranked_by_score=True)
+
+    scores = get_label_quality_scores(labels, predictions)
+
+    print(f" Number of images with issue{len(label_issue_idx)}")
+    print(f"Average quality score {np.mean(scores)}")
+
+    if data_dir:
+        for idx in label_issue_idx[:5]:
+            label = labels[idx]
+            prediction = predictions[idx]
+            score = scores[idx]
+            image_path = data_dir + label['seg_map']  # Path to the image file
+
+            print(f"Image: {image_path} | Label quality score: {score} | Is issue: True")
+            visualize(image_path, label=label, prediction=prediction, class_names=classes_mapping, overlay=False)
+
+
+if __name__ == "__main__":
+    coco_annotations_path = '/Users/alexuvarovskiy/Documents/mlp_dataset/data/result.json'
+    preds_path = '/Users/alexuvarovskiy/Documents/mlp/annotation/predictions.json'
+    classes_mapping = {0: "person", 1: "car"}
```

Comment:
What are the results of the check? Could you update the README with them, please?